spyderlib/widgets/arraybuilder.py

```diff
@@ -140,8 +143,9 @@ def text(self):
                 else:
                     value = '0'
 
-                if value == '':
+                if value.strip() == '':

```

Comment:
`if not value.strip():` maybe ? (this is just style, but since you're changing it...)